File: java/Samples/Biometrics/VB/EnrollmentSampleVB/Forms/DeviceSelectForm.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
Imports Neurotec.Biometrics
Imports Neurotec.Devices

Namespace Forms
	Partial Public Class DeviceSelectForm
		Inherits Form
		#Region "Public constructor"

		Public Sub New()
			InitializeComponent()
		End Sub

		#End Region

		#Region "Public properties"

		Private privateDeviceManager As NDeviceManager
		Public Property DeviceManager() As NDeviceManager
			Get
				Return privateDeviceManager
			End Get
			Set(ByVal value As NDeviceManager)
				privateDeviceManager = value
			End Set
		End Property

		Private privateSelectedDevice As NFScanner
		Public Property SelectedDevice() As NFScanner
			Get
				Return privateSelectedDevice
			End Get
			Set(ByVal value As NFScanner)
				privateSelectedDevice = value
			End Set
		End Property

		#End Region

		#Region "Private form methods"

		Private Sub ScannerFormLoad(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			cbScanner.BeginUpdate()
			Try
				cbScanner.Items.Clear()
				If DeviceManager IsNot Nothing Then
					For Each item As NDevice In DeviceManager.Devices
						If TypeOf item Is NFScanner Then
							cbScanner.Items.Add(item)
						End If
					Next item

					cbScanner.SelectedItem = SelectedDevice
					If cbScanner.SelectedItem Is Nothing AndAlso cbScanner.Items.Count > 0 Then
						cbScanner.SelectedIndex = 0
					End If
				End If
			Finally
				cbScanner.EndUpdate()
			End Try
		End Sub

		Private Sub CbScannerSelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles cbScanner.SelectedIndexChanged
			SelectedDevice = TryCast(cbScanner.SelectedItem, NFScanner)
			If SelectedDevice IsNot Nothing Then
				Dim canCaptureSlaps As Boolean = False
				Dim canCaptureRolled As Boolean = False

				For Each item As NFPosition In SelectedDevice.GetSupportedPositions()
					If (Not NBiometricTypes.IsPositionFourFingers(item)) Then
						Continue For
					End If
					canCaptureSlaps = True
					Exit For
				Next item

				For Each item As NFImpressionType In SelectedDevice.GetSupportedImpressionTypes()
					If (Not NBiometricTypes.IsImpressionTypeRolled(item)) Then
						Continue For
					End If
					canCaptureRolled = True
					Exit For
				Next item

				If canCaptureRolled Then
					lblCanCaptureRolled.Image = Global.Neurotec.Samples.My.Resources.Ok
				Else
					lblCanCaptureRolled.Image = Global.Neurotec.Samples.My.Resources.Bad
				End If

				If canCaptureSlaps Then
					lblCanCaptureSlaps.Image = Global.Neurotec.Samples.My.Resources.Ok
				Else
					lblCanCaptureSlaps.Image = Global.Neurotec.Samples.My.Resources.Bad
				End If
			Else
				lblCanCaptureRolled.Visible = False
				lblCanCaptureSlaps.Visible = False
			End If
		End Sub

		Private Sub BtnOkClick(ByVal sender As Object, ByVal e As EventArgs) Handles btnOk.Click
			If SelectedDevice IsNot Nothing Then
				DialogResult = System.Windows.Forms.DialogResult.OK
			Else
				Utilities.ShowWarning("Scanner not selected")
			End If
		End Sub

		#End Region
	End Class
End Namespace

File: java/Samples/Biometrics/CS/EnrollmentSampleCS/Forms/DeviceSelectForm.cs
using System;
using System.Windows.Forms;
using Neurotec.Biometrics;
using Neurotec.Devices;
using Neurotec.Samples.Properties;

namespace Neurotec.Samples.Forms
{
	public partial class DeviceSelectForm : Form
	{
		#region Public constructor

		public DeviceSelectForm()
		{
			InitializeComponent();
		}

		#endregion

		#region Public properties

		public NDeviceManager DeviceManager { get; set; }

		public NFScanner SelectedDevice { get; set; }

		#endregion

		#region Private form methods

		private void ScannerFormLoad(object sender, EventArgs e)
		{
			cbScanner.BeginUpdate();
			try
			{
				cbScanner.Items.Clear();
				if (DeviceManager != null)
				{
					foreach (NDevice item in DeviceManager.Devices)
					{
						if (item is NFScanner)
						{
							cbScanner.Items.Add(item);
						}
					}

					cbScanner.SelectedItem = SelectedDevice;
					if (cbScanner.SelectedItem == null && cbScanner.Items.Count > 0)
					{
						cbScanner.SelectedIndex = 0;
					}
				}
			}
			finally
			{
				cbScanner.EndUpdate();
			}
		}

		private void CbScannerSelectedIndexChanged(object sender, EventArgs e)
		{
			SelectedDevice = cbScanner.SelectedItem as NFScanner;
			if (SelectedDevice != null)
			{
				bool canCaptureSlaps = false;
				bool canCaptureRolled = false;

				foreach (NFPosition item in SelectedDevice.GetSupportedPositions())
				{
					if (!NBiometricTypes.IsPositionFourFingers(item)) continue;
					canCaptureSlaps = true;
					break;
				}

				foreach (NFImpressionType item in SelectedDevice.GetSupportedImpressionTypes())
				{
					if (!NBiometricTypes.IsImpressionTypeRolled(item)) continue;
					canCaptureRolled = true;
					break;
				}

				if (canCaptureRolled) lblCanCaptureRolled.Image = Resources.Ok;
				else lblCanCaptureRolled.Image = Resources.Bad;

				if (canCaptureSlaps) lblCanCaptureSlaps.Image = Resources.Ok;
				else lblCanCaptureSlaps.Image = Resources.Bad;
			}
			else
			{
				lblCanCaptureRolled.Visible = false;
				lblCanCaptureSlaps.Visible = false;
			}
		}

		private void BtnOkClick(object sender, EventArgs e)
		{
			if (SelectedDevice != null) DialogResult = DialogResult.OK;
			else
				Utilities.ShowWarning("Scanner not selected");
		}

		#endregion
	}
}
